src/interactionManager/interactionManager.ts

```diff
@@ -16,6 +16,7 @@ flows.forEach((f) => {
 export interface InteractionEvents {
   interactionCreated: (interxn: Interaction) => void
   interactionUpdated: (interxn: Interaction) => void
+  interactionResumed: (interxn: Interaction) => void
```

Comment:
this can be abstracted into
```
type InteractionEventKeys = "interactionCreated" | "interactionUpdated" | "interactionResumed"
export type InteractionEvents = {
  [key in InteractionEventKeys]: (interxn: Interaction) => void
}

Reply:
or even more interesting
```
type InteractionEventKeys = "create" | "update" | "resume"
export type InteractionEvents = {
  [key in InteractionEventKeys as `interaction${Capitalize<key>}d`]: (interxn: string) => void
}

Reply:
> or even more interesting
> 
> ```
> type InteractionEventKeys = "create" | "update" | "resume"
> export type InteractionEvents = {
>   [key in InteractionEventKeys as `interaction${Capitalize<key>}d`]: (interxn: Interaction) => void
> }
> ```

I think this approach could cost a problems with refactoring since finding in project entrances like this is impossible, you must for sure know where and what to look for...